profiles/car.lua

```diff
@@ -390,6 +392,74 @@ function process_way(profile, way, result)
   }
 
   WayHandlers.run(profile,way,result,data,handlers)
+
+  -- now process relations data
+  local matched_refs = nil;
+  if result.ref then
+    local match_res = Relations.match_to_ref(relations, result.ref)
+
+    local ref = ''
+    for _, m in pairs(match_res) do
+      if ref ~= '' then
+        ref = ref .. '; '
+      end
+
+      if m.dir then
+        ref = ref .. m.ref .. ' $' .. m.dir
```

Comment:
This style of appending a cardinal direction to a route number is specified by the [MUTCD](https://mutcd.fhwa.dot.gov/htm/2009r1r2/part2/part2e.htm#section2E17) (used in the U.S., Canada, and a few other regions), but [`direction` is used all over the world](https://taginfo.openstreetmap.org/keys/direction#map). I’m not sure if it’s used on road route relations in non-MUTCD countries, or if it would be surprising for users to see cardinal directions appended in those countries.

Reusing the raw relation role or `direction` tag in the human-readable `ref` also assumes that the instruction will ultimately be given in English. If the user is viewing a route in Québec in French, they would expect [this way](http://www.openstreetmap.org/way/470714815) to have a ref of `35 Nord; 104 Ouest`, not `35 North; 104 West`. It would make sense for OSRM Text Instructions to be responsible for translating the cardinal direction into the right language, but that would entail some iffy text replacement unless the ref is also provided in a structured format (#3329).